File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.cpp
class Solution {
public:
   long long int fact(int c){
        long long int sum=0;
        while(c){
            sum=sum+c;
            c--;
        }
        return sum;
    }
    long long zeroFilledSubarray(vector<int>& nums) {
        long long int c=0;
            long long int sum=0;
            for(long long int n: nums){
                if (n==0)
                    c++;
                else{
                    if (c!=0){
                        sum=sum+(fact(c));
                        c=0;
                        }
                    }
                }
            if (c!=0)
                 sum=sum+(fact(c));
        return sum;
    }
};

File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.py
class Solution:
   
    def zeroFilledSubarray(self, nums: List[int]) -> int: 
            def fact(c):
                sum=0
                while c:
                   sum=sum+c
                   c-=1
                return sum
            c=0
            sum=0
            for n in nums:
                if n==0:
                    c+=1
                else:
                    if c!=0:
                        sum=sum+(fact(c))
                        c=0
            if c!=0:
                 sum=sum+(fact(c))
            return sum